### PR TITLE
INT-3615: @MessagingGatewy Long Properties

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INT-3615

Change timeout properties from long to String to allow
property placeholder/SpEL configuration.

### CODE REVIEW

File: spring-integration-core/src/main/java/org/springframework/integration/annotation/MessagingGateway.java [hunk 1]
```diff
@@ -68,7 +68,7 @@
 	 * For example if this gateway is hooked up to a {@code QueueChannel}. 
 	 * @return the suggested timeout in milliseconds, if any
 	 */
-	long defaultRequestTimeout() default Long.MIN_VALUE;
+	String defaultRequestTimeout() default "9223372036854775807";
```
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/annotation/MessagingGateway.java]: ???
This isn't `Long.MIN_VALUE`...

**Comment by artembilan** [comment]
So, aren't you going to change `@Gateway`, since we know already how to overcome `Long.MIN_VALUE`?

**Comment by garyrussell** [comment]
We'd have to add SpEL/PP support to the GPFB.

Let's wait until someone asks - we have more important things to work on.

File: spring-integration-core/src/main/java/org/springframework/integration/annotation/MessagingGateway.java (hunk 1 above)
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/annotation/MessagingGateway.java]: OK. Just let me know, if you don't have anything specific to this point.
And I'll fix on merge or leave as is.
Just convince me that this change is correct or not :smile:

Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/annotation/MessagingGateway.java]: Doh it's MAX, lol

**Comment by artembilan** [comment]
Merging with mentioned fix...

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/029bb28cfa5ca2ddeadeac11b687486f1c8d9aba

**Pull request closed without merging**